### PR TITLE
membership: Restructure pallet into separate files

### PR DESCRIPTION
**Pull request by vatsa287**
- What does this PR do?
   This PR separates `membership` pallet into separate files for `lib`, `mock` & `tests`.
   
- Why are these changes needed?
   Currently `membership` pallet consists of `lib`, `mock` & `tests` written into a single file.
   Refactor it into separate files which makes it inline to other pallets and improves readability.

- How were these changes implemented and what do they affect?
   The PR will not have any affect.

Signed-off-by: Shreevatsa N <i.mnshreevatsa@gmail.com>

### CODE REVIEW

**Comment by vatsa287** [comment]
Label required: `T1-Frame`.
Not able to add this into the PR.

**Comment by xlc** [comment]
you forget to commit the lib.rs file?

**Comment by vatsa287** [comment]
> you forget to commit the lib.rs file?

Sorry yes. I have updated the commit now. 
Thanks.

File: substrate/frame/membership/src/benchmark.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+// This file is part of Substrate.
```
Comment (ggwpez) [inline substrate/frame/membership/src/benchmark.rs:1]: We normally call this file `benchmarking.rs` by convention.

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/membership/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,225 @@
+// This file is part of Substrate.
```
Comment (ggwpez) [inline substrate/frame/membership/src/tests.rs:1]: The files are never used?

**Comment by ggwpez** [comment]
Please check that everything still works when running the tests.

File: substrate/frame/membership/src/tests.rs (hunk 2 above)
Comment (vatsa287) [reply substrate/frame/membership/src/tests.rs:1]: Had missed this.
Added these files into `lib.rs`.

File: substrate/frame/membership/src/benchmark.rs (hunk 1 above)
Comment (vatsa287) [reply substrate/frame/membership/src/benchmark.rs]: Ack.
Updated to `benchmarking.rs`.

**Comment by vatsa287** [comment]
> Please check that everything still works when running the tests.

Yes it works fine on my local setup.
```rust
vatsa@Shreevatsas-MacBook-Pro polkadot-sdk % cargo test -p pallet-membership --features=runtime-benchmarks
    Finished `test` profile [unoptimized + debuginfo] target(s) in 1.60s
     Running unittests src/lib.rs (target/debug/deps/pallet_membership-f44be527ad6bc1b4)

running 22 tests
test mock::__construct_runtime_integrity_test::runtime_integrity_tests ... ok
test mock::test_genesis_config_builds ... ok
test benchmarking::bench_clear_prime ... ok
test benchmarking::bench_set_prime ... ok
test benchmarking::bench_swap_member ... ok
test tests::add_member_works ... ok
test benchmarking::bench_change_key ... ok
test benchmarking::bench_reset_members ... ok
test benchmarking::bench_remove_member ... ok
test benchmarking::bench_add_member ... ok
test tests::change_key_with_same_caller_as_argument_changes_nothing ... ok
test tests::change_key_works_that_does_not_change_order ... ok
test tests::change_key_works ... ok
test tests::query_membership_works ... ok
test tests::swap_member_with_identical_arguments_changes_nothing ... ok
test tests::prime_member_works ... ok
test tests::remove_member_works ... ok
test tests::reset_members_works ... ok
test tests::swap_member_works ... ok
test tests::swap_member_works_that_does_not_change_order ... ok
test tests::migration_v4 ... ok
test tests::genesis_build_panics_with_duplicate_members - should panic ... ok

test result: ok. 22 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s

   Doc-tests pallet-membership

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

File: substrate/frame/membership/src/lib.rs [hunk 3]
```diff
@@ -33,6 +33,15 @@ use sp_std::prelude::*;
 pub mod migrations;
 pub mod weights;
 
+#[cfg(any(feature = "mock", test))]
```
Comment (ggwpez) [inline substrate/frame/membership/src/lib.rs:36]: Does this `mock` feature exist? I dont think its needed otherwise.

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/membership/src/lib.rs (hunk 3 above)
Comment (vatsa287) [reply substrate/frame/membership/src/lib.rs:36]: Yes mock is the file where `mock` runtime is defined.

**Comment by vatsa287** [comment]
@ggwpez any other changes required? Could you please approve the workflow.
Thanks.

File: substrate/frame/membership/src/lib.rs [hunk 4]
```diff
@@ -33,6 +33,15 @@ use sp_std::prelude::*;
...
+pub mod mock;
```
Comment (bkchr) [inline substrate/frame/membership/src/lib.rs:37]:
```suggestion
#[cfg(test)]
mod mock;
```

File: substrate/frame/membership/src/tests.rs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+use sp_runtime::{bounded_vec, traits::BadOrigin, BuildStorage};
+
+use frame_support::{
+    assert_noop, assert_ok, assert_storage_noop, 
+    traits::{StorageVersion},
+};
+use frame_support::traits::PalletInfo;
+
+#[test]
+	fn query_membership_works() {
```
Comment (bkchr) [inline substrate/frame/membership/src/tests.rs:32]:
```suggestion
fn query_membership_works() {
```

File: substrate/frame/membership/src/tests.rs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        phantom: Default::default(),
+    }
+    .build_storage()
+    .unwrap();
+}
+
+#[test]
+fn migration_v4() {
+    new_test_ext().execute_with(|| {
+        //use frame_support::traits::PalletInfo;
```
Comment (bkchr) [inline substrate/frame/membership/src/tests.rs:209]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Comment by vatsa287** [comment]
@ggwpez @bkchr 
Thanks for the help!

**Pull request merged**